File: LeetCode/Java Solutions/Longest Subarray With Maximum Bitwise AND/longest.java
class Solution {
    public int longestSubarray(int[] nums) {
        int maxNum = 0;
        for (int num : nums) {
            maxNum = Math.max(maxNum, num);
        }
        
        int currentLength = 0;
        int maxLength = 0;
        
        for (int num : nums) {
            if (num == maxNum) {
                currentLength++;
                maxLength = Math.max(maxLength, currentLength);
            } else {
                currentLength = 0;
            }
        }
        
        return maxLength;
    }
}

File: LeetCode/Python Solutions/Longest Subarray With Maximum Bitwise AND/longest.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        max_num = max(nums)
        current_length = max_length = 0
        
        for num in nums:
            if num == max_num:
                current_length += 1
                max_length = max(max_length, current_length)
            else:
                current_length = 0
        
        return max_length